From a2d496f1a441563d2a0ea96a1b5fecf6c0d4f901 Mon Sep 17 00:00:00 2001
Message-Id: <a2d496f1a441563d2a0ea96a1b5fecf6c0d4f901.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Akash Goel <akash.goel@intel.com>
Date: Tue, 10 Jun 2014 16:16:14 +0530
Subject: [PATCH 113/312] FOR_UPSTREAM [VPG]: drm/i915: Disable Turbo if Rpe
 equals Rp0

On VLV, if Rpe=Rp0, which means both Rpe and max frequency are same
and so Graphics system will run at constant frequency only, hence
turbo logic & corresponding interrupts are not needed.

Issue: VIZ-3936
Change-Id: Iee1bc7c122e5b186c1d6784b31e4b51e4eac11ab
Signed-off-by: Deepak S <deepak.s@intel.com>
Signed-off-by: Akash Goel <akash.goel@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 3e718df..2c4ca2f 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4337,7 +4337,11 @@ static void valleyview_enable_rps(struct drm_device *dev)
 	/* enable WA for RC6+turbo to work together */
 	dev_priv->rps.use_RC0_residency_for_turbo = true;
 
-	vlv_set_rps_mode(dev, false);
+	/* If Rpe=Rp0, disable turbo to avoid interrupt processing overhead. */
+	if (dev_priv->rps.efficient_freq == dev_priv->rps.max_freq)
+		vlv_set_rps_mode(dev, true);
+	else
+		vlv_set_rps_mode(dev, false);
 
 	gen6_gt_force_wake_put(dev_priv, FORCEWAKE_ALL);
 }
-- 
1.7.9.5

